File: python-systempromptleakage/secure7.py
import hashlib
import logging
from openai import OpenAI

logger = logging.getLogger(__name__)
client = OpenAI()

system_prompt = "Internal assistant. Confidential instructions including escalation paths."

def process(user_input: str) -> str:
    prompt_hash = hashlib.sha256(system_prompt.encode()).hexdigest()[:8]
    logger.info("Processing with prompt hash=%s length=%d", prompt_hash, len(system_prompt))
    response = client.chat.completions.create(
        model="gpt-4o",
        max_tokens=1024,
        messages=[
            {"role": "system", "content": system_prompt},
            {"role": "user", "content": user_input}
        ]
    )
    return response.choices[0].message.content

File: typescript-sensitiveinfodisclosure/secure7.ts
import OpenAI from "openai";
import { Request } from "express";

const client = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

async function handleUserQuery(req: Request): Promise<string> {
  const userQuestion: string = req.body.question;
  const response = await client.chat.completions.create({
    model: "gpt-4o",
    max_tokens: 1024,
    messages: [
      { role: "user", content: userQuestion }
    ],
  });
  return response.choices[0].message.content ?? "";
}
